### PR TITLE
fix #25271: Part export automatic numeration of same instruments

### LINKED ISSUE #25271: "Edit element" option improvements
### Issue type

See the title.

### Description with steps to reproduce

Few ideas to improve "Edit element" option when right-clicking something on the score:

(from @MarcSabatella comment)
I'd be very much in favor of seeing some useful indication that you are in edit mode and what it does. Like, a popup that appears showing clicking arrow buttons (in addition to enabling the cursor keys). Or even just some additional highlighting of the element being edited (a colored background, say).

(Editing this report if someone encounter some problem).

### What is the latest version of MuseScore Studio where this issue is present?

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.2-242570931, revision: github-musescore-musescore-3130f97

### Regression

I was unable to check

### Operating system

Windows 10 Pro

### Additional context

_No response_

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

### PR DESCRIPTION
**Pull request by vinayakvivek**
Implemented a function _**renameWithNumbers**_ which takes in a ExcerptList and renames properly.
part-export

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Looks good to me

**Comment by Jojo-Schmitz** [comment]
Looks good to me, Hopefully would make it into 2.1 too

**Comment by vinayakvivek** [comment]
@Jojo-Schmitz thanks :)

File: libmscore/excerpt.cpp [hunk 1]
```diff
@@ -1142,13 +1142,49 @@ QList<Excerpt*> Excerpt::createAllExcerpt(MasterScore *score)
                   for (int i = part->startTrack(), j = 0; i < part->endTrack(); i++, j++) {
                         e->tracks().insert(i, j);
                         }
-                  QString name = createName(part->partName(), all);
+                  QString name = part->partName();
```
Comment (Jojo-Schmitz) [inline libmscore/excerpt.cpp]: This might be a bad change, as `createName()` calls `partName.simplified()` to cleanup spaces?

Comment (vinayakvivek) [inline libmscore/excerpt.cpp]: so `name = part->partName().simplified()` would be ok. right?

Comment (Jojo-Schmitz) [inline libmscore/excerpt.cpp]: I think so

Comment (vinayakvivek) [inline libmscore/excerpt.cpp]: @Jojo-Schmitz thanks for the review :)

Comment (Jojo-Schmitz) [inline libmscore/excerpt.cpp]: Well, why not use the original `createName()` and fix that to work the same way as your new `renameWithNumbers()` ?

Comment (vinayakvivek) [inline libmscore/excerpt.cpp]: but the function `ExcerptsDialog::createName(const QString& partName)` calls `Excerpt::createName(const QString& partName, QList<Excerpt*> excerptList)`

Comment (Jojo-Schmitz) [inline libmscore/excerpt.cpp]: `Excerpt::createAllExcerpt()` and `ExcerptsDialog::createName()` are the only callers of `Excerpt::createName()`, both should benefit from the automatic numbering I think?

File: libmscore/excerpt.cpp [hunk 2]
```diff
@@ -1154,22 +1154,26 @@ QList<Excerpt*> Excerpt::createAllExcerpt(MasterScore *score)
...
-            for (int k = 0; k < count; ++k) {
-                  ee = excerptList[k];
-                  if (ee->title() == name)
-                        break;
-                  }
-            if ((ee == 0) || (ee->title() != name))
+      QString name = partName.simplified();
+      int count = 0;    // no of occurences of partName
+      
+      foreach(Excerpt* e, excerptList) {
```
Comment (Jojo-Schmitz) [inline libmscore/excerpt.cpp]: better use `for (Excerpt* e : excerptLlist)`

File: libmscore/excerpt.cpp [hunk 3]
```diff
@@ -1154,22 +1154,26 @@ QList<Excerpt*> Excerpt::createAllExcerpt(MasterScore *score)
...
+            // if <partName> already exists, change <partName> to <partName 1>
+            if (e->title().compare(name) == 0) {
+                  e->setTitle(e->title() + QString(" %1").arg("1"));
+                  count = 1;
                   break;
+                  }
+          
+            if (e->title().mid(0, e->title().size() - 2) == name)
```
Comment (lasconic) [inline libmscore/excerpt.cpp]: That will not work for more than 10 parts right?

File: libmscore/excerpt.cpp [hunk 4]
```diff
@@ -1154,22 +1154,26 @@ QList<Excerpt*> Excerpt::createAllExcerpt(MasterScore *score)
...
-                        break;
-                  }
-            if ((ee == 0) || (ee->title() != name))
+      QString name = partName.simplified();
+      int count = 0;    // no of occurences of partName
+      
+      foreach(Excerpt* e, excerptList) {
+            // if <partName> already exists, change <partName> to <partName 1>
+            if (e->title().compare(name) == 0) {
+                  e->setTitle(e->title() + QString(" %1").arg("1"));
```
Comment (lasconic) [inline libmscore/excerpt.cpp]: `QString(" %1").arg("1")`--> `" 1"` ?

Comment (Jojo-Schmitz) [inline libmscore/excerpt.cpp]: Hmm, maybe better `QString(" %1").arg(tr("1"))`?

File: libmscore/excerpt.cpp (hunk 3 above)
Comment (vinayakvivek) [inline libmscore/excerpt.cpp]: @lasconic didn't get the problem with this. Could you pls explain the issue?

File: libmscore/excerpt.cpp (hunk 2 above)
Comment (vinayakvivek) [inline libmscore/excerpt.cpp]: ok.. will change it

File: libmscore/excerpt.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [inline libmscore/excerpt.cpp]: Maybe not, won't work further down

File: libmscore/excerpt.cpp (hunk 3 above)
Comment (vinayakvivek) [inline libmscore/excerpt.cpp]: Works for any number of parts now :)
screen shot 2017-02-16 at 10 11 12 pm

**Comment by IsaacWeiss** [comment]
Is there anything holding up the merge of this?

**Pull request merged**